src/django/api/services/opensearch/search.py

```diff
@@ -46,11 +46,22 @@ def __prepare_opensearch_response(self, response):
             else:
                 logger.warning(f"Missing '_source' in hit: {hit}")
 
-        return {
+        response_data = {
             "count": total_hits,
-            "data": data
+            "data": data,
         }
 
+        buckets = (
+            response.get("aggregations", {})
+            .get("grouped", {})
+            .get("buckets", [])
+        )
+
+        if buckets:
+            response_data["aggregation_data"] = buckets
```

Comment:
I would suggest changing the response structure a little bit. When we add the aggregation to the response the way we are doing it in the code now, that implies the following response structure:
```json
{
  "data", [...],
  "aggregation_data": [...]
}
```

I my opinion that limits us in terms of response schema, so that the `aggregation_data` is basically the same as `buckets` in OpenSearch response.
I would suggest doing something like this instead:
```json
{
  "data": [...],
  "aggregations": {
    "geohex_grid": [...], 
  }
}
```
That would open us up for the future aggregation and creates a pattern that other aggregations can follow.